PokemonWorkshop/PokemonStudio
# Implement new tooltip

This PR branches out of Select branch since it kind of depends on it. It'll be de-branched-out as soon as Select get reviewed.

## Description

This PR adds tooltip that automatically show when mouse hovers an element that:

- Has text ellipsis overflow
- Wants to show a tooltip when being hovered (via dataset attributes)
- Wants to show a fancy tooltip when being hovered (via dataset attributes)

Note: The data attributes are wrong but will be fixed later. I'll update the page as soon as I'm able to safely update this PR with the right data attributes (aka. whenever the select PR is merged).

Note2: I'm not removing legacy tooltip yet for the same reason: select PR isn't merged.

## Tests to perform

- Open the Design System. 
- Go to Tooltip, test all shown case (don't forget to scroll down a little to check that tooltip show on bottom when they can't do it on top).
- Go to Select, open the Dialog, check that the tooltip option actually shows the tooltip in dialog as well ;)
